packages/nimble-components/src/table-column/duration-text/models/duration-formatter.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * A class for formatting a numeric value in the unit of milliseconds into a display value representing its
+ *  days, hours, minutes, and seconds.
+ */
+export class DurationFormatter {
+    private readonly daysFormatter: Intl.NumberFormat;
+    private readonly hoursFormatter: Intl.NumberFormat;
+    private readonly minutesFormatter: Intl.NumberFormat;
+    private readonly secondsFormatter: Intl.NumberFormat;
+    private readonly scientificSecondsFormatter: Intl.NumberFormat;
+
+    public constructor(private readonly lang: string) {
+        this.daysFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'day'
+        });
+        this.hoursFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'hour'
+        });
+        this.minutesFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'minute'
+        });
+        this.secondsFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'second'
+        });
+        this.scientificSecondsFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'second',
+            notation: 'scientific',
+            maximumFractionDigits: 3
+        });
+    }
+
+    public format(milliseconds: number | null | undefined): string {
+        if (
+            milliseconds === null
+            || milliseconds === undefined
+            || milliseconds < 0
+            || !Number.isFinite(milliseconds)
+        ) {
+            return '';
+        }
+
+        const result = [];
+        const millisecondsPerSecond = 1000;
+        const millisecondsPerMinute = millisecondsPerSecond * 60;
+        const millisecondsPerHour = millisecondsPerMinute * 60;
+        const millisecondsPerDay = millisecondsPerHour * 24;
+        const fractionalDays = milliseconds / millisecondsPerDay;
+        let remainingTime = milliseconds;
+        const days = Math.floor(fractionalDays);
+        if (days <= 100 && days > 0) {
+            const formattedDays = this.daysFormatter.format(days);
+            result.push(formattedDays);
+            remainingTime -= days * millisecondsPerDay;
+        } else if (days > 100) {
```

Comment:
Was there a discussion on where 100 days came from? Maybe we wait for feedback and don't reopen now but I think durations in the couple year category would be pretty possible and "600 days" would be more useful than the value in seconds. I'd advocate for a value like 999 (the rollover to 4 digits) as the big number to abandon ship.

Reply:
Sold!